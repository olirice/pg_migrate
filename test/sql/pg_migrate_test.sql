-- Create dependencies
create extension "uuid-ossp";


-- Create pg_migrate 
create extension pg_migrate;


-- Select from upgrade table
select stmt from migrations.statement;


-- current_statement_id
begin;
    create function test_fn () returns bool as $$ select true $$ language sql;
    select migrations.current_statement_id() is not null;
rollback;


-- current_statement
begin;
    create function test_fn () returns bool as $$ select true $$ language sql;
    select (migrations.current_statement()).id is not null;
rollback;


-- TODO: register a downgrade
-- TODO: cut revision from statemetns
-- TODO: cut revision & apply downgrade
-- TODO: cut revision, apply downgarde, test revision
-- TODO: full migration cylce integration test
