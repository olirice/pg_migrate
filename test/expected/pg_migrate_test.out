-- Create dependencies
create extension "uuid-ossp";
-- Create pg_migrate 
create extension pg_migrate;
-- Select from upgrade table
select stmt from migrations.statement;
 stmt 
------
(0 rows)

-- Get revision_id
begin;
    create function test_fn () returns bool as $$ select true $$ language sql;
INFO:  Detected CREATE FUNCTION create function test_fn () returns bool as $$ select true $$ language sql;
    select migrations.current_statement_id() is not null;
 ?column? 
----------
 t
(1 row)

rollback;
-- Get revision
begin;
    create function test_fn () returns bool as $$ select true $$ language sql;
INFO:  Detected CREATE FUNCTION create function test_fn () returns bool as $$ select true $$ language sql;
    select (migrations.current_statement()).id is not null;
 ?column? 
----------
 t
(1 row)

rollback;
-- TODO: register a downgrade
-- TODO: cut revision from statemetns
-- TODO: cut revision & apply downgrade
-- TODO: cut revision, apply downgarde, test revision
-- TODO: full migration cylce integration test
